advisor/src/main/kotlin/advisors/Osv.kt

```diff
@@ -0,0 +1,199 @@
+                    )
+                )
+            } else {
+                emptyList()
+            }
+        }
+    }
+
+    private fun getVulnerabilitiesForPackage(packages: List<Package>): Map<Identifier, List<Vulnerability>> {
+        val vulnerabilityIdsForPackageId = getVulnerabilityIdsForPackages(packages)
+        val allVulnerabilityIds = vulnerabilityIdsForPackageId.values.flatten().toSet()
+        val vulnerabilityForId = getVulnerabilitiesForIds(allVulnerabilityIds).associateBy { it.id }
+
+        return packages.associate { pkg ->
+            pkg.id to vulnerabilityIdsForPackageId[pkg.id].orEmpty().map { vulnerabilityForId.getValue(it) }
+        }
+    }
+
+    private fun getVulnerabilityIdsForPackages(packages: List<Package>): Map<Identifier, List<String>> {
+        val requests = packages.mapNotNull { pkg ->
+            val request = createRequest(pkg) ?: return@mapNotNull null
+            pkg to request
+        }
+
+        val result = service.getVulnerabilityIdsForPackages(requests.map { it.second })
+        if (result.isFailure) {
+            log.error {
+                "Requesting vulnerabilities IDs for packages failed: ${result.exceptionOrNull()!!.collectMessages()}"
+            }
+        }
+
+        return buildMap {
+            result.getOrNull().orEmpty().forEachIndexed { i, vulnerabilities ->
+                put(requests[i].first.id, vulnerabilities)
+            }
+
+            packages.forEach { pkg ->
+                putIfAbsent(pkg.id, emptyList())
+            }
+        }
+    }
+
+    private fun getVulnerabilitiesForIds(ids: Set<String>): List<Vulnerability> {
+        val result = service.getVulnerabilitiesForIds(ids)
+
+        return result.getOrNull() ?: run {
+            log.error {
+                "Requesting vulnerabilities IDs for packages failed: ${result.exceptionOrNull()!!.collectMessages()}"
+            }
+            emptyList()
+        }
+    }
+}
+
+private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest? {
+    val name: String? = when {
+        pkg.id.namespace.isEmpty() -> pkg.id.name
+        pkg.id.type == "Composer" -> "${pkg.id.namespace}/${pkg.id.name}"
+        else -> "${pkg.id.namespace}:${pkg.id.name}"
+    }
+
+    val ecosystem: String? = when (pkg.id.type) {
+        "Bower" -> null
+        "Composer" -> Ecosystem.PACKAGIST
+        "Crate" -> Ecosystem.CRATES_IO
+        // TODO: The entries for Go should be merged, see https://github.com/oss-review-toolkit/ort/issues/5533.
+        "GoDep" -> Ecosystem.GO
+        "GoMod" -> Ecosystem.GO
+        "NPM" -> Ecosystem.NPM
+        "NuGet" -> Ecosystem.NUGET
+        "Maven" -> Ecosystem.MAVEN
+        "PyPI" -> Ecosystem.PYPI
+        else -> null
+    }
+
+    // Strip the version prefix for GoMod / GoDep packages. That prefix should not be assigned by the package manager
+    // in the first place, see https://github.com/oss-review-toolkit/ort/issues/5532.
```

Comment:
How about fixing that issue first, so we don't need to temporarily add this code?

Reply:
As this is a breaking change which requires our user to make alignments in their internal setup for which the PR would need to wait, I prefer to do it afterwards. Will you accept that?